# From: Dimitry Andric <dimitry@andric.com>
# Date: 2004-05-27
#
# Description:
#
# This patch fixes <sys/sysctl.h> so it doesn't cause several Linux
# kernel headers to be ignored, even if you WANT them to be compiled.
#
# See http://sources.redhat.com/ml/libc-hacker/2004-04/msg00097.html
# and http://sources.redhat.com/cgi-bin/cvsweb.cgi/libc/sysdeps/unix/sysv/linux/sys/sysctl.h.diff?r1=1.6&r2=1.7&cvsroot=glibc

diff -urNd glibc-2.3.2-ct/sysdeps/unix/sysv/linux/sys/sysctl.h glibc-2.3.2/sysdeps/unix/sysv/linux/sys/sysctl.h
--- glibc-2.3.2-ct/sysdeps/unix/sysv/linux/sys/sysctl.h	2002-11-06 06:43:44.000000000 +0100
+++ glibc-2.3.2/sysdeps/unix/sysv/linux/sys/sysctl.h	2004-05-27 16:30:47.000000000 +0200
@@ -1,4 +1,4 @@
-/* Copyright (C) 1996, 1999, 2002 Free Software Foundation, Inc.
+/* Copyright (C) 1996, 1999, 2002, 2003, 2004 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
 
    The GNU C Library is free software; you can redistribute it and/or
@@ -23,11 +23,44 @@
 #define __need_size_t
 #include <stddef.h>
 /* Prevent more kernel headers than necessary to be included.  */
-#define _LINUX_KERNEL_H	1
-#define _LINUX_TYPES_H	1
-#define _LINUX_LIST_H	1
+#ifndef _LINUX_KERNEL_H
+# define _LINUX_KERNEL_H	1
+# define __undef_LINUX_KERNEL_H
+#endif
+#ifndef _LINUX_TYPES_H
+# define _LINUX_TYPES_H		1
+# define __undef_LINUX_TYPES_H
+#endif
+#ifndef _LINUX_LIST_H
+# define _LINUX_LIST_H		1
+# define __undef_LINUX_LIST_H
+#endif
+#ifndef __LINUX_COMPILER_H
+# define __LINUX_COMPILER_H	1
+# define __user
+# define __undef__LINUX_COMPILER_H
+#endif
+
 #include <linux/sysctl.h>
 
+#ifdef __undef_LINUX_KERNEL_H
+# undef _LINUX_KERNEL_H
+# undef __undef_LINUX_KERNEL_H
+#endif
+#ifdef __undef_LINUX_TYPES_H
+# undef _LINUX_TYPES_H
+# undef __undef_LINUX_TYPES_H
+#endif
+#ifdef __undef_LINUX_LIST_H
+# undef _LINUX_LIST_H
+# undef __undef_LINUX_LIST_H
+#endif
+#ifdef __undef__LINUX_COMPILER_H
+# undef __LINUX_COMPILER_H
+# undef __user
+# undef __undef__LINUX_COMPILER_H
+#endif
+
 __BEGIN_DECLS
 
 /* Read or write system parameters.  */
